host/sgx/windows/debugrtbridge.c

```diff
@@ -0,0 +1,132 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#include <Windows.h>
+#include <openenclave/internal/debugrt/host.h>
+#include <openenclave/internal/trace.h>
+#include "../../hostthread.h"
+
+static struct
+{
+    oe_once_type once;
+    HMODULE hmodule;
+    oe_result_t (*notify_enclave_created)(oe_debug_enclave_t* enclave);
+    oe_result_t (*notify_enclave_terminated)(oe_debug_enclave_t* enclave);
+    oe_result_t (*push_thread_binding)(
+        oe_debug_enclave_t* enclave,
+        struct _sgx_tcs* tcs);
+    oe_result_t (*pop_thread_binding)(void);
+} _oedebugrt;
+
+static void get_debugrt_function(const char* name, FARPROC* out)
+{
+    *out = GetProcAddress(_oedebugrt.hmodule, name);
+    if (*out == NULL)
+    {
+        OE_TRACE_FATAL("Could not find function %s in oedebugrt.dll", name);
+    }
+}
+
+static void load_oedebugrt(void)
+{
+    if (_oedebugrt.hmodule != NULL)
+    {
+        OE_TRACE_FATAL("oedebugrt.dll has already been loaded. Multiple load "
+                       "attempts detected.");
+    }
+
+#ifndef NDEBUG
+    /**
+     *  In debug mode, give preference to OE_DEBUGRT_PATH.
+     *  This is mainly used for OE SDK development.
+     */
+    {
+        char* debugrtpath = getenv("OE_DEBUGRT_PATH");
+        _oedebugrt.hmodule = LoadLibraryExA(
+            debugrtpath,
+            NULL, /* reserved */
+            /* Search only specified path. */
+            LOAD_LIBRARY_SEARCH_DEFAULT_DIRS);
+    }
+
+#endif
+
+    /* Search for oedebugrt.dll only in the application folder. */
+    if (_oedebugrt.hmodule == NULL)
+    {
+        _oedebugrt.hmodule = LoadLibraryExA(
+            "oedebugrt.dll",
+            NULL, /* reserved */
+            LOAD_LIBRARY_SEARCH_APPLICATION_DIR);
+    }
```

Comment:
Here we are automatically loading the debug extension? How does this relate with someone using the .load command to load the extension?

Reply:
Also can we unload the dell we have loaded?

Reply:
> Also can we unload the dll we have loaded?

Done.

Reply:
> Here we are automatically loading the debug extension? How does this relate with someone using the .load command to load the extension?

We are loading the debugger runtime. The runtime will send notifications (enclave create/terminate etc) to the debugger (if one is attached).
If the debug extension has been loaded (via the .load command or by someother means), then the notifications will be handled.